egret/models/unit_commitment.py

```diff
@@ -763,6 +763,11 @@ def _lazy_ptdf_termination_cleanup(m, md, time_periods, solver, ptdf_options, pr
         if persistent_solver:
             lpu._load_pf_slacks(solver, m, t_subset)
         _lazy_ptdf_warmstart_copy_violations(m, md, time_periods, solver, ptdf_options, prepend_str)
+        # After adding so many constraints,
+        # the warmstart basis is typically
+        # worse than useless. If the solver
+        # implements it, reset the solver.
+        solver.reset()
```

Comment:
Won't this cause an error for anything other than the gurobi persistent interface?

Reply:
I thought so for the longest time as well, but this is actually a method on the `OptSolver` base class:
https://github.com/Pyomo/pyomo/blob/5.7.1/pyomo/opt/base/solvers.py#L752-L756

It would be better if the other persistent solvers implemented this method, but obviously that needs to be updated in Pyomo. In any case, it doesn't seem to do any harm on solvers that aren't gurobi persistent.

Reply:
Interesting...

Reply:
I actually did not know that existed, haha!